lib/requestChecks.js

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var deepEqual = require('deep-equal')
+
+module.exports = function (config, mc, log) {
+
+  var requestChecks = null
+  var pollInterval = null
+
+  function RequestChecks(settings) {
+    this.setAll(settings)
+  }
+
+  RequestChecks.prototype.setAll = function (settings) {
+    this.treatEveryoneWithSuspicion = settings.treatEveryoneWithSuspicion
+    // The private branch puts some additional private config here.
+    return this
+  }
+
+  RequestChecks.prototype.push = function () {
+    log.info({ op: 'requestChecks.push' })
+    return mc.setAsync('requestChecks', this, 0)
+      .then(this.refresh.bind(this))
+  }
+
+  RequestChecks.prototype.refresh = function (options) {
+    log.info({ op: 'requestChecks.refresh' })
+    var result = mc.getAsync('requestChecks').then(validate)
+
+    if (options && options.pushOnMissing) {
+      result = result.catch(this.push.bind(this))
+    }
+
+    return result.then(
+      this.setAll.bind(this),
+      function (err) {
+        log.error({ op: 'requestChecks.refresh', err: err })
+      }
+    )
+  }
+
+  RequestChecks.prototype.pollForUpdates = function () {
+    this.stopPolling()
+    pollInterval = setInterval(this.refresh.bind(this), config.updatePollIntervalSeconds * 1000)
+  }
+
+  RequestChecks.prototype.stopPolling = function () {
+    clearInterval(pollInterval)
+  }
+
+  function validate(settings) {
```

Comment:
nit: This function seems to do a little more than just validate. Possibly add comment to mention that it returns updated settings based on `config.requestChecks`?